hack/install.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# Inspired by - https://github.com/vmware-tanzu/community-edition/blob/main/hack/install.sh
+# Script to install tanzu framework
+# Usage: ./hack/install.sh /path/to/tanzu-framework-binary v0.10.0
+
+# set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+ALLOW_INSTALL_AS_ROOT="${ALLOW_INSTALL_AS_ROOT:-""}"
+if [[ "$EUID" -eq 0 && "${ALLOW_INSTALL_AS_ROOT}" != "true" ]]; then
+  echo "Do not run this script as root"
+  exit 1
+fi
+
+MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+ARCH="amd64"
+
+case "${OS}" in
+  linux)
+    XDG_DATA_HOME="${HOME}/.local/share"
+    ;;
+  darwin)
+    XDG_DATA_HOME="${HOME}/Library/Application Support"
+    ;;
+  *)
+    echo "${OS} is unsupported"
+    exit 1
+    ;;
+esac
+echo "${XDG_DATA_HOME}"
+
+# check if the tanzu CLI already exists and remove it to avoid conflicts
+TANZU_BIN_PATH=$(command -v tanzu)
+if [[ -n "${TANZU_BIN_PATH}" ]]; then
+  # best effort, so just ignore errors
+  sudo rm -f "${TANZU_BIN_PATH}" > /dev/null
+fi
+
+# set install dir to /usr/local/bin
+TANZU_BIN_PATH="/usr/local/bin"
+echo Installing tanzu cli to "${TANZU_BIN_PATH}"
+
+# if plugin cache pre-exists, remove it so new plugins are detected
+TANZU_PLUGIN_CACHE="${HOME}/.cache/tanzu/catalog.yaml"
+if [[ -n "${TANZU_PLUGIN_CACHE}" ]]; then
+  echo "Removing old plugin cache from ${TANZU_PLUGIN_CACHE}"
+  rm -f "${TANZU_PLUGIN_CACHE}" > /dev/null
+fi
+
+# install tanzu cli
+install $1/cli/core/$2/tanzu-core-"${OS}"_"${ARCH}" /usr/local/bin/tanzu
```

Comment:
- on Linux, `install` command would require `sudo` AFAIK
- use `TANZU_BIN_PATH` for destination

Reply:
I have addressed this in https://github.com/vmware-tanzu/tanzu-framework/pull/1157/commits/db0e4d47f9654288c1a2e9893e069783e1df4580 

thanks!